pandas/core/base.py

```diff
@@ -713,9 +713,17 @@ def argmax(self, axis=None, skipna: bool = True, *args, **kwargs) -> int:
         the minimum cereal calories is the first element,
         since series is zero-indexed.
         """
+        delegate = self._values
         nv.validate_minmax_axis(axis)
-        nv.validate_argmax_with_skipna(skipna, args, kwargs)
-        return nanops.nanargmax(self._values, skipna=skipna)
+        skipna = nv.validate_argmax_with_skipna(skipna, args, kwargs)
+
+        if isinstance(delegate, ExtensionArray):
+            if not skipna and isna(delegate).any():
```

Comment:
why are you doing EA logic here? I would rather pass the EA skipna and have it handle it

Reply:
The current EA implementation of argmax/min does not handle skipna. Do you suggest I should put the skipna logic there instead?

Reply:
I think EA is designed to be similar to np.array, that's why the currently implementation of EA.argmax doesn't handle skipna, just like np.array.argmax. For the np case argmax is wrapped by nanops.argmax that handles skipna. For the EA case since the skipna logic is stright forward I thought I could just write it out without doing a wrapping like nanops.argmax.

Alternatively we can totally do what you suggest i.e. let EA.argmax handle skipna. I don't have an opinion.

Reply:
Ideally the EA.argmax/argmin do handle `skipna`, but the main problem is that `Series.argmax/argmin` has very peculiar NA handling that we probably don't want to copy to EA, so we need to keep this custom check here anyway to preserve the Series behaviour. 
See the two issues linked in https://github.com/pandas-dev/pandas/issues/35178 about those topics.

But we could also change the series behaviour for the *new* EA dtypes

Reply:
@jorisvandenbossche I agree it's probably best to split into two cases: one for EA for the regular np array.
@jreback I have now moved the skipna logic to EA's argin/max implementation